R/teal_slice.R

```diff
@@ -336,36 +336,63 @@ is.teal_slices <- function(x) { # nolint
 }
 
 
-# JSON utils ------------------------------------------------------------------------------------------------------
+# utils -----------------------------------------------------------------------------------------------------------
 
-slice_to_json <- function(x) {
+trim_character <- function(x, nchar) {
+  x_t <- substr(x, 1, nchar)
+  substr(x_t, nchar - 2, nchar) <- "..."
+  x_t
+}
+
+justify_json_split <- function(json, format_fun) {
+  json_s <- strsplit(json, split = ":", fixed = TRUE)
+  name_width <- max(unlist(gregexpr(":", json))) - 1
+  vapply(json_s, function(x) paste0(format_fun(x[1], name_width), na.omit(x[2])), character(1))
+}
+
+# JSON utils for teal_slice ---------------------------------------------------------------------------------------
+
+slice_list_to_json <- function(x) {
+  # selected and choices should remain arrays
   if (!is.null(x$selected)) x$selected <- I(x$selected)
   if (!is.null(x$choices)) x$choices <- I(x$choices)
 
   jsonlite::toJSON(x, pretty = TRUE, auto_unbox = TRUE, digits = 16, null = "null")
 }
 
-trim_character <- function(x, nchar) {
-  json_t <- substr(x, 1, nchar)
-  substr(json_t, nchar - 2, 40) <- "..."
-  json_t
+slice_format_name <- function(n, name_width) {
+  if (nchar(n) == 1) {
+    return(n)
+  } else {
+    paste(format(n, width = name_width), ":")
+  }
 }
 
-center_json_split <- function(json) {
-  format_name <- function(n) {
-    if (nchar(n) == 1) {
-      return(n)
-    } else {
-      paste(format(n, width = name_width), ":")
-    }
-  }
+# JSON utils for teal_slices --------------------------------------------------------------------------------------
 
-  json_s <- strsplit(json, split = ":", fixed = TRUE)
-  name_width <- max(unlist(gregexpr(":", json))) - 1
+slices_list_to_json <- function(x) {
+
+  # selected and choices should remain arrays
+  x$slices <- lapply(x$slices, function(x) {
+    if (!is.null(x$selected)) x$selected <- I(x$selected)
+    if (!is.null(x$choices)) x$choices <- I(x$choices)
+    x
+  })
+
+  jsonlite::toJSON(x, pretty = TRUE, auto_unbox = TRUE, digits = 16, null = "null")
+}
```

Comment:
If we drop `auto_unbox` then we could remove `slices_list_to_json` and `slice_list_to_json`

Reply:
changed in d5d353b229f99ce138887f3d1a0d52fd4b680a5b